tests/integration/test_sns.py

```diff
@@ -2211,7 +2211,8 @@ def test_subscription_after_failure_to_deliver(
         message = "test_dlq_after_sqs_endpoint_deleted"
         sns_client.publish(TopicArn=topic_arn, Message=message)
         # to avoid race condition, publish is async and the redrive policy can be in effect before the actual publish
-        time.sleep(1)
+        # there isn't a proper way to check that the message has been published before setting the RedrivePolicy
+        time.sleep(3)
```

Comment:
could we create a second subscription to a queue that we poll for checking whether the message has arrived?

Reply:
Yes, that could help, but it would not guarantee that the other message was published yet, even though it greatly reduce the chance that this would happen.

Reply:
I've done it that way now. Only thing is, AWS added a new field to subscriptions (`SubscriptionPrincipal`) and it creates a new `<resource:id>` in the snapshots. We will need to regenerate the snapshots and implement it at some point.